core/templates/domain/topic_viewer/read-only-topic-object.factory.ts

```diff
@@ -121,40 +135,35 @@ export class ReadOnlyTopicObjectFactory {
     this._subtopicObjectFactory = subtopicObjectFactory;
     this._skillSummaryObjectFactory = skillSummaryObjectFactory;
   }
-  // TODO(#7176): Replace 'any' with the exact type. This has been kept as
-  // 'any' because 'topicDataDict' is a dict with underscore_cased keys
-  // which give tslint errors against underscore_casing in favor of camelCasing.
-  createFromBackendDict(topicDataDict: any): ReadOnlyTopic {
-    let subtopics: Array<Subtopic> = topicDataDict.subtopics.map(
-      (subtopic: ISubtopicBackendDict) => {
-        return this._subtopicObjectFactory.create(
-          subtopic, topicDataDict.skill_descriptions);
-      });
+
+  createFromBackendDict(
+      topicDataDict: IReadOnlyTopicBackendDict): ReadOnlyTopic {
+    let subtopics: Array<Subtopic> = topicDataDict.subtopics.map(subtopic => {
```

Comment:
Please replace all usages of `Array<T>` with `T[]` like so:

```suggestion
    let subtopics: Subtopic[] = topicDataDict.subtopics.map(subtopic => {
```

Reply:
done